policy/modules/system/authlogin.te

```diff
@@ -555,6 +555,10 @@ optional_policy(`
 	samba_dontaudit_write_var_files(nsswitch_domain)
 ')
 
+optional_policy(`
+	systemd_userdbd_rw_data(nsswitch_domain)
```

Comment:
Only root can write to /run/systemd/userdb so this is too broad

There are three sock files maintained in there:
* dynamicuser. this is used by nss_systemd (nsswtch_domain) to connect to systemd (init_t) with a unix stream socket.
* multiplexer: this is used by pam clients (programs that has a pam config in /etc/pam.d)  to connect to userdbd with a unix domain socket
* home: this is used by nss_systemd (nsswitch_domain) to connect to homed with a unix stream socket (if home is enabled)

Reply:
@doverride  Are you suggesting to label each socket with different label?

Reply:
I don't do that myself no but should be possible. You can differentiate using the "connectto" permission:

* init_t:unix_stream_socket connectto (for dynamicuser)
* userdbd_t:unix_stream_socket connectto (for multiplexer)
* homed_t::unix_stream_socket connectto (for home)

Reply:
from `man userdbctl`
```
WELL-KNOWN SERVICES
       The userdbctl services command will list all currently running services that provide user or group definitions to the system. The
       following are well-known services are shown among this list.

       io.systemd.DynamicUser
           This service is provided by the system service manager itself (i.e. PID 1) and makes all users (and their groups) synthesized
           through the DynamicUser= setting in service unit files available to the system (see systemd.exec(5) for details about this setting).

       io.systemd.Home
           This service is provided by systemd-homed.service(8) and makes all users (and their groups) belonging to home directories managed by
           that service available to the system.

       io.systemd.Multiplexer
           This service is provided by systemd-userdbd.service(8) and multiplexes user/group look-ups to all other running lookup services.
           This is the primary entry point for user/group record clients, as it simplifies client side implementation substantially since they
           can ask a single service for lookups instead of asking all running services in parallel.  userdbctl uses this service preferably,
           too, unless --with-nss= or --service= are used, in which case finer control over the services to talk to is required.

       io.systemd.NameSeviceSwitch
           This service is (also) provided by systemd-userdbd.service(8) and converts classic NSS/glibc user and group records to JSON
           user/group records, providing full backwards compatibility. Use --with-nss=no to disable this compatibility, see above. Note that
           compatibility is actually provided in both directions: nss-systemd(8) will automatically synthesize classic NSS/glibc user/group
           records from all JSON user/group records provided to the system, thus using both APIs is mostly equivalent and provides access to
           the same data, however the NSS/glibc APIs necessarily expose a more reduced set of fields only.

       Note that userdbctl has internal support for NSS-based lookups too. This means that if neither io.systemd.Multiplexer nor
       io.systemd.NameSeviceSwitch are running look-ups into the the basic user/group databases will still work.
```

Reply:
note that the "homed" service can be optional. You dont have to have systemd-homed enabled.

Reply:
Okay, I'll keep it as we have it now. Fedora policy should not be so strict.